doc/string.md

```diff
@@ -39,13 +33,41 @@ operator std::u16string() const;
 ```
 Converts a String value to a UTF-16 encoded C++ string.
 
+## Methods
+
+### New
+```cpp
+String::New();
+```
+Creates a new empty String
+
+### New
+```cpp
+String::New(napi_env env, const std::string& value);
+String::New(napi_env env, const std::u16string& value);
+String::New(napi_env env, const char* value);
+String::New(napi_env env, const char16_t* value);
```

Comment:
We should talk about this.  Its not consistent with anything I've seen elsewhere in the node documentation

Reply:
But with overrides it does make sense to me

Reply:
Yes, that was kind of what I was thinking. Anyone else have thoughts?